catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/RegistryPublicationServiceImpl.java

```diff
@@ -0,0 +1,211 @@
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.registry.federationadmin.service.impl;
+
+import java.io.Serializable;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.registry.api.RegistryStore;
+import org.codice.ddf.registry.common.metacard.RegistryObjectMetacardType;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminException;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminService;
+import org.codice.ddf.registry.federationadmin.service.RegistryPublicationService;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+
+public class RegistryPublicationServiceImpl implements RegistryPublicationService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegistryPublicationService.class);
+
+    private FederationAdminService federationAdminService;
+
+    private Map<String, RegistryStore> registryStoresByRegistryId = new ConcurrentHashMap<>();
+
+    @Override
+    public void publish(String registryId, String destinationRegistryId)
+            throws FederationAdminException {
+        Metacard metacard = getMetacard(registryId);
+        String metacardId = metacard.getId();
+
+        List<Serializable> locations = new ArrayList<>();
+        Attribute locAttr = metacard.getAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS);
+        if (locAttr != null) {
+            locations = new HashSet<>(locAttr.getValues()).stream()
+                    .map(Object::toString)
+                    .collect(Collectors.toCollection(ArrayList::new));
+        }
+
+        if (locations.contains(destinationRegistryId)) {
+            return;
+        }
+
+        String sourceId = getSourceIdFromRegistryId(destinationRegistryId).orElse(null);
+
+        if (StringUtils.isNotBlank(sourceId)) {
```

Comment:
Could this if just be the body of an 'ifPresent' on the optional returned above? Don't know if that makes sense or not